apps/mobile/src/components/search/index.tsx

```diff
@@ -41,11 +72,33 @@ const SearchComponent = ({searchQuery, setSearchQuery}: ISearchComponent) => {
       </View>
       <TextInput
         style={styles.input}
-        value={searchQuery}
+        value={debouncedQuery}
         onChangeText={handleChangeText}
         placeholder="Search"
         clearButtonMode="always"
       />
+      <Pressable onPress={() => setIsOpenFilter(true)}>
+        <Text style={styles.input}>Filter</Text>
+      </Pressable>
+
+      <FilterMenu
+        visible={isOpenFilter}
+        onClose={() => setIsOpenFilter(false)}
+        kinds={kinds}
+        setKinds={setKinds}
+        onSortChange={handleSortChange}
+        activeSortBy={activeSortBy}
+      />
+
+      {isLoading && <Text>Loading...</Text>}
+      {isError && <Text>Error loading data</Text>}
+      {data?.pages?.map((page, index) => (
```

Comment:
We have already a Flat list, you need to add the filter and search on the Feed not here